nightly-playground/lib/common-tools-stack.ts

```diff
@@ -7,25 +7,60 @@ compatible open source license. */
 
 import { Stack, StackProps } from 'aws-cdk-lib';
 import { Certificate, CertificateValidation } from 'aws-cdk-lib/aws-certificatemanager';
+import {
+  Effect, ManagedPolicy, PolicyStatement, Role, ServicePrincipal,
+} from 'aws-cdk-lib/aws-iam';
 import { HostedZone } from 'aws-cdk-lib/aws-route53';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
 import { Construct } from 'constructs';
 
 export class CommonToolsStack extends Stack {
-    readonly certificateArn: string
+  readonly certificateArn: string
 
-    readonly zone = 'playground.nightly.opensearch.org'
+  public readonly customRole: Role
 
-    constructor(scope: Construct, id: string, props: StackProps) {
-      super(scope, id, props);
+  readonly zone = 'playground.nightly.opensearch.org'
 
-      const route53HostedZone = new HostedZone(this, 'nigghhtlyHostedZone', {
-        zoneName: this.zone,
-      });
+  constructor(scope: Construct, id: string, props: StackProps) {
+    super(scope, id, props);
 
-      const certificate = new Certificate(this, 'cert', {
-        domainName: this.zone,
-        validation: CertificateValidation.fromDns(route53HostedZone),
-      });
-      this.certificateArn = certificate.certificateArn;
-    }
+    const route53HostedZone = new HostedZone(this, 'nigghhtlyHostedZone', {
+      zoneName: this.zone,
+    });
+
+    const certificate = new Certificate(this, 'cert', {
+      domainName: this.zone,
+      validation: CertificateValidation.fromDns(route53HostedZone),
+    });
+    this.certificateArn = certificate.certificateArn;
+
+    this.customRole = new Role(this, 'customInstanceRole', {
+      managedPolicies: [ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ReadOnlyAccess'),
+        ManagedPolicy.fromAwsManagedPolicyName('CloudWatchAgentServerPolicy'),
+        ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore')],
+      assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+    });
+
+    const snapshotS3Bucket = new Bucket(this, 'snapshotS3Bucket', {
+      bucketName: 'nightly-playgrounds-snapshots-bucket',
+    });
+
+    const s3bucketPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        's3:ListBucket',
```

Comment:
Do we really need these many actions?
ListBucket will show too much for the S3 bucket as well as GetBucketLocation?

Reply:
Got from here: https://github.com/opensearch-project/opensearch-cluster-cdk/blob/main/lib/infra/remote-store-resources.ts#L17-L32
On our website it says to give `s3:*` which in my opinion is way to risky